scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -731,12 +731,30 @@ object FormatWriter {
     }
   }
 
-  private val trailingSpace = Pattern.compile("\\h+$", Pattern.MULTILINE)
+  /**
+    * No Unicode is necessary in the three (3) Pattern.compile() below.
+    * scalafmt parser will have already have objected.
```

Comment:
are you saying that nothing other than space and tab are allowed by scalac or scalameta parser that precedes formatting?

Reply:
It **shouldn't** be allowed according to specs. 

A cursory examination of the scalameta sources led to [this](https://github.com/scalameta/scalameta/blob/0fcba272a9f8d91b22fb8a180a805acf443a30be/scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/Chars.scala#L99). So, it matches.

Reply:
After another long day of creating test cases and trying to be perverse, it turns out
that trailing whitespace with Unicode whitespace _is_ allowed in comments. As cited,
it is rejected in non-comment code.

I am about to do a commit which handles, tests, & succinctly (I hope) documents.

The discussion here helps.